src/services/pending-queue.ts

```diff
@@ -191,16 +194,40 @@ export async function cancelEditingPendingQueueSubmission(
  *
  * @returns
  */
-export function getPendingQueueSubmissions(): Promise<PendingFormSubmission[]> {
-  return utilsService.localForage
-    .getItem('submissions')
-    .then((submissions) => (Array.isArray(submissions) ? submissions : []))
-}
+export async function getPendingQueueSubmissions(): Promise<
+  PendingFormSubmission[]
+> {
+  const pendingQueueSubmissionsInStorage =
+    await utilsService.getLocalForageItem<PendingFormSubmission[]>(
+      PENDING_QUEUE_SUBMISSIONS_KEY,
+    )
 
-export function getFormSubmission(
-  pendingTimestamp: string,
-): Promise<FormSubmission | null> {
-  return utilsService.getLocalForageItem(`SUBMISSION_${pendingTimestamp}`)
+  if (!pendingQueueSubmissionsInStorage) {
+    return []
+  }
+
+  // Form submission data use to be stored separate to the array of pending records
+  // so to ensure all pending records have submission data we will pull it from
+  // where it use to be stored if there is isn't any form submission data.
+  // This is could only happen if a user had the latest code (I.e. they are online)
+  // and they have a submission in the pending queue that cannot be submitted
+  // (E.g. they need to login or one of the attachments could not be uploaded).
+  const pendingQueueSubmissions: PendingFormSubmission[] = []
```

Comment:
this may never be reached because of line 205 (unless I'm mistaken)

Reply:
If there is nothing in storage, there is nothing to iterate over.